platform/linux-generic/include/odp/api/plat/packet_inlines.h

```diff
@@ -126,6 +131,63 @@ static inline uint32_t _odp_packet_l4_offset(odp_packet_t pkt)
 	return _odp_pkt_get(pkt, uint16_t, l4_offset);
 }
 
+/** @internal Inline function @param pkt @param len @return */
+static inline void *_odp_packet_l2_ptr(odp_packet_t pkt, uint32_t *len)
+{
+	uint32_t offset  = _odp_packet_l2_offset(pkt);
+	uint32_t seg_len = _odp_packet_seg_len(pkt);
+	uint8_t *data    = (uint8_t *)_odp_packet_data(pkt);
```

Comment:
Could you please check if moving a call to `_odp_packet_data()` after `if` will speed things?

Reply:
The fast path here is not calling _odp_packet_map(), but return data+offset. So, data is likely needed anyway and in theory better to load it as soon as possible, so that offset and len checks can be done while waiting for data read. However in reality the placement of the C line does not matter much, since GCC find optimal place for the data read call and out-of-order CPU also tries to parallelize the read with the other checks.

So, it think C line placement of _odp_packet_data() does not make a difference here.

Reply:
@psavol ok.